pages/edge-services/reference-content/understanding-pricing.mdx

```diff
@@ -173,4 +206,50 @@ The example prices and limits used below are subject to change. You should alway
-e.g. `0.0135 * 200 = €2.70`
\ No newline at end of file
+e.g. `0.0135 * 200 = €2.70`
+
+### WAF charges
+
+Any additional WAF request charges accumulated when you exceeded your previous plan's limit will remain on your monthly bill. From the moment you change your plan, your WAF request consumption resets to 0. During the rest of the month, you can consume WAF requests up to the new plan's limit without being charged.  
+
+<Message type="note">
+- If you **downgrade** your plan, all the WAF requests made within the hour of changing plans will count towards the WAF request consumption of the new plan. 
+- If you **upgrade** your plan, all the WAF requests made within the hour of changing plans will count towards the cache consumption of the old plan.
+</Message>
+
+For example:
+
+<Message type="important">
+The example prices and limits used below are subject to change. You should always refer to the [pricing page](https://www.scaleway.com/en/pricing/network/#edge-services) for the most up to date information.
+</Message>
+
+- From November 1-10 you are subscribed to the **Professional** plan, which has a limit of 5M WAF requests.
+- From November 1-10 you make 10M WAF requests.
+- On November 11, you upgrade to the **Advanced** plan, which has a limit of 50M WAF requests. Your WAF request usage resets to 0.
+- Between November 11 and the end of the month, you make 50M WAF requests.
+
+You made 5M WAF requests that were not included within your monthly Professional plan, between November 1-10. For the rest of the month, you were within the limits of your new Advanced plan. Your November Edge Services billing, in terms of the additional cache charges, is therefore calculated as follows:
+
+`{Fee per additional 1M WAF requests} * 5` <br/>
+e.g. `0.5* 5 = €2.50`
+
+### WAF add-on
+
+This scenario applies to a user who is subscribed to the Starter plan and pays the WAF add-on price, then upgrades to a higher level plan.
+
+You will be billed pro-rata for the add-on charge, based on how long you were subscribed Starter with an add-on before upgrading, during the billing month. 
+
+For example:
+
+<Message type="important">
+The example prices and limits used below are subject to change. You should always refer to the [pricing page](https://www.scaleway.com/en/pricing/network/#edge-services) for the most up to date information.
+</Message>
+
+- From November 1-10 you are subscribed to the **Starter** plan and **WAF add-on** 
+- On November 11, you switch to the **Professional** plan.
+- You remain subscribed to the **Professional** plan for the rest of the month (ending November 30)
+
+There are 30 days in the month of November, and you spent 10 of them subscribed to the Starter plan with the WAF add-on. When you upgrade to the Professional plan, the WAF add-on was no longer applicable because WAF was automatically included in your subscription plan. Your November Edge Services billing, in terms of the WAF add-on, is therefore calculated as follows:
```

Comment:
```suggestion
There are 30 days in the month of November, and you spent 10 of them subscribed to the Starter plan with the WAF add-on. When you upgraded to the Professional plan, the WAF add-on was no longer applicable because WAF was automatically included in your subscription plan. Your November Edge Services billing, in terms of the WAF add-on, is therefore calculated as follows:
```